src/femtomes.h

```diff
@@ -0,0 +1,217 @@
+
+/**
+* Analysis Femto uavgps frame header
+*/
+typedef union {
+	femto_msg_header_t 	femto_header;
+	uint8_t 	 	data[28];
+}msg_header_t;
+
+/**
+* receive Femto complete uavgps frame
+*/
+typedef struct {
+	uint8_t 		data[256];		/**< receive Frame message content */
+	uint32_t 		crc;			/**< receive Frame message crc 4 bytes */
+	msg_header_t 		header;			/**< receive Frame message header */
+	uint16_t 		read;			/**< receive Frame message read bytes count */
+} femto_msg_t;
+
+#pragma pack(pop)
+/*** END OF femtomes protocol binary message and payload definitions ***/
+
+enum class FemtoDecodeState {
+	pream_ble1,			/**< Frame header preamble first byte 0xaa */
+	pream_ble2,			/**< Frame header preamble second byte 0x44 */
+	pream_ble3,			/**< Frame header preamble third byte 0x12 */
+	head_length,			/**< Frame header length */
+	head_data,			/**< Frame header data */
+	data,				/**< Frame data */
+	crc1,				/**< Frame crc1 */
+	crc2,				/**< Frame crc2 */
+	crc3,				/**< Frame crc3 */
+	crc4				/**< Frame crc4 */
+};
+
+class GPSDriverFemto : public GPSBaseStationSupport
+{
+public:
+	/**
+	 * @param heading_offset heading offset in radians [-pi, pi]. It is substracted from the measurement.
+	 */
+	GPSDriverFemto(GPSCallbackPtr callback, void *callback_user, struct vehicle_gps_position_s *gps_position,
+			 struct satellite_info_s *satellite_info, float heading_offset = 0.f);
+	virtual ~GPSDriverFemto();
+
+	int receive(unsigned timeout) override;
+	int configure(unsigned &baudrate, OutputMode output_mode) override;
+
+private:
+
+	/**
+	 * caculate the frame crc value
+	 */
+    	uint32_t crc32Value(uint32_t icrc);
+    	uint32_t calculateBlockCRC32(uint32_t length, uint8_t *buffer, uint32_t crc);
+
+	/**
+	 * when Constructor is work, initialize parameters
+	 */
+	void decodeInit(void);
+
+	/**
+	 * check the message if whether is 8001,memcpy data to _gps_position
+	 */
+	int handleMessage(int len);
+
+	/**
+	 * analysis frame data from buf[] to _femto_msg and check the frame is suceess or not
+	 */
+	int parseChar(uint8_t b);
+
+	/**
+	 * Write a command and wait for a (N)Ack
+	 * @return 0 on success, <0 otherwise
+	 */
+	int writeAckedCommandFemto(const char* command, const char* reply, const unsigned timeout);
+
+	/**
+	 * receive data for at least the specified amount of time
+	 */
+	void receiveWait(unsigned timeout_min);
+
+	/**
+	 * enable output of correction output
+	 */
+	void activateCorrectionOutput();
+
+	void sendSurveyInStatusUpdate(bool active, bool valid, double latitude = (double)NAN, double longitude = (double)NAN,
+				      float altitude = NAN);
+
+	void activateRTCMOutput();
+
+	struct satellite_info_s *_satellite_info {nullptr};
+	struct vehicle_gps_position_s *_gps_position {nullptr};
+	femto_uav_gps_t _femto_uav_gps;
+	femto_msg_t _femto_msg;
+	FemtoDecodeState _decode_state{FemtoDecodeState::pream_ble1};
+
+	bool _got_pashr_pos_message{false}; /**< If we got a PASHR,POS message, we will ignore GGA messages */

```

Comment:
`_got_pashr_pos_message` is still unused.

Reply:
thanks，i have delete all not used variables and functions